### PR TITLE
Add Ruby environment for release tools

### PR DESCRIPTION
**Pull request by loremattei**
This PR adds a Ruby environment to the image. Ruby is used by most of our release tooling. 
Ruby is installed via `rbenv` as different repositories still use different versions of Ruby. 
By default, the image ships with `2.7.4` which currently is the most common.

Note that the Debian repository has a `rbenv` package, but it's pretty old and the versions of Ruby we currently use are not available. Since `ruby-tool` is not installed as a `rbenv` plugin in the standard package, but as a binary in a different package, updating it would require adding a different Debian repository. Because of this, I preferred to keep things clean and simple and install `rbenv` from the source repository.

### CODE REVIEW

File: Dockerfile
Comment (loremattei) [inline Dockerfile]: This is the [suggested build environment](https://github.com/rbenv/ruby-build/wiki#suggested-build-environment) for `rbenv`. 
A couple of packages are already installed in `Debian`, but since there's no arm with keeping all of them in the command line, I preferred to stick with the complete set.

**Review by oguzkocer — APPROVED** [review]
I don't have the ruby/rbenv setup instructions memorized, but this all looks reasonable to me 😅 I don't think there is too much value in going through the documentation and checking each line, so I am going to add a couple other notes about the PR instead:

1. @jkmassel was working on WCAndroid Buildkite integration which might have some overlap with this one. I don't see any branches related to it, but pinging him to avoid possibly duplicating further effort.
2. This increases the image size from `1.02GB` to `1.52GB`. I don't think that's a negligible increase, but I don't think we need to block the PR either. Since the images are version controlled, when we are planning to update the image for other repositories, we can run some tests to decide whether it'd be worth splitting the image into two, with & without the Ruby tooling.
3. @loremattei In case you weren't aware, we push the Buildkite image for PRs, so you can use `public.ecr.aws/automattic/android-build-image:4281c9e97b2d821df3de34c046b7c067499b35bb` to test this PR.
4. With (2) in mind, assuming you want to merge this PR, I am not sure if we should tag the new image `v1.2.0` or something else. What we do matters less than the fact that we are communicating it with the rest of the team, so we don't upgrade the other projects to the new version without understanding its implications.

Hope these notes help and thank you for implementing this! 🙇‍♂️

**Comment by loremattei** [comment]
Yep! Thank you for your notes @oguzkocer. 
50% increase in the image size is not really negligible. I'll keep the PR open and I'll test the current implementation of installable build that I have and that relies on Ruby. 

I'm thinking if we should do something different on Android. AFAICT we can do almost everything with just `gradle`. We would just need to implement commenting on GitHub wtithout Ruby and move ahead with the standard image. 
Wdyt?

**Comment by oguzkocer** [comment]
@loremattei I am not familiar with DayOne which I am guessing what this PR is for, but in other client projects if I am not mistaken we'd need the ruby toolchain for fastlane.

**Comment by loremattei** [comment]
Yep, that's right. But, on Android, we can implement many tasks without fastlane. Fastlane would probably be still very useful for some stuff (like screenshot automation). But we may think about having two different images and use the one with Ruby only for the tasks that need it. 
Obviously, the downside is to have to maintain both of them. 
It's probably worth to experiment and measure the impact of the bigger image in order to make an informed decision.

**Comment by oguzkocer** [comment]
> Yep, that's right. But, on Android, we can implement many tasks without fastlane. 

All our release-toolkit related implementation is in `fastlane`, are you suggesting for us to switch some of them over?

> But we may think about having two different images and use the one with Ruby only for the tasks that need it.

Yeap, I suggested that as an option a couple times, including in this review in (2): `Since the images are version controlled, when we are planning to update the image for other repositories, we can run some tests to decide whether it'd be worth splitting the image into two, with & without the Ruby tooling.`

**Pull request closed without merging**
